qcodes/loops.py

```diff
@@ -249,22 +250,25 @@ def each(self, *actions):
         return ActiveLoop(self.sweep_values, self.delay, *actions,
                           then_actions=self.then_actions, station=self.station,
                           progress_interval=self.progress_interval,
-                          bg_task=self.bg_task, bg_min_delay=self.bg_min_delay)
+                          bg_task=self.bg_task, end_task=self.end_task, bg_min_delay=self.bg_min_delay)
 
-    def with_bg_task(self, task, min_delay=1):
+    def with_bg_task(self, task, end_task=None, min_delay=1):
```

Comment:
@giulioungaretti I think this works fine, but I'm not sure the two features (background task and end task) should be joined at the hip like this. If we think of this end_task as specifically finishing whatever was done by the background task, which seems reasonable to me, then linking them is fine, but perhaps a more descriptive name for end_task would help make it clearer to the user. bg_final_task? bg_task_finish? end_bg_task?